src/cas/spreadsheet_to_cas.py

```diff
@@ -34,18 +35,39 @@ def read_spreadsheet(file_path: str, sheet_name: Optional[str]):
     Returns:
         tuple: Tuple containing metadata (dict), column names (list), and raw data (pd.DataFrame).
     """
-    # Read the specific sheet or the first sheet if sheet_name is not provided
+    schema_file = resources.files(schemas) / "CAP_schema.json"
+    with schema_file.open("rt") as f:
+        cap_schema = json.loads(f.read())
     if sheet_name:
         spreadsheet_df = pd.read_excel(file_path, sheet_name=sheet_name, header=None)
     else:
         spreadsheet_df = pd.read_excel(file_path, header=None)
-    # Extract meta data (first 8 rows)
-    meta_data = dict(spreadsheet_df.iloc[:8, 0:2].to_records(index=False))
-    # Extract column names (9th row)
-    column_names = spreadsheet_df.iloc[8, :].tolist()
-    # Extract raw data (from 10th row onwards)
-    raw_data = spreadsheet_df.iloc[9:, :]
-    raw_data.columns = column_names
+
+    meta_data = {}
+    column_names = []
+    raw_data = pd.DataFrame()
+
+    header_row_index = None
+    for index, row in spreadsheet_df.iterrows():
+        first_cell = str(row[0])
+        if first_cell.startswith("#"):
+            key = first_cell[1:].strip()
+            value = row[1] if pd.notnull(row[1]) else ""
+            if key in cap_schema["properties"]:
+                meta_data[key] = value
```

Comment:
`author_name`, `orcid` and `author_list` is not populated. Also we should get type of the field from schema and parse accordingly. Value of `author_list` should be list not string.